cmd/src/gateway_benchmark.go

```diff
@@ -55,26 +58,49 @@ Examples:
 			return cmderrors.Usage("additional arguments not allowed")
 		}
 
-		// Create HTTP client with TLS skip verify
-		client := &http.Client{Transport: &http.Transport{}}
-
-		endpoints := map[string]string{
-			"HTTP":                fmt.Sprintf("%s/gateway", cfg.Endpoint),
-			"HTTP then WebSocket": fmt.Sprintf("%s/gateway/http-then-websocket", cfg.Endpoint),
+		var (
+			gatewayWebsocket, sourcegraphWebsocket *websocket.Conn
+			err                                    error
+			httpClient                             = &http.Client{}
+			endpoints                              = map[string]any{}
+		)
+		if *gatewayEndpoint != "" {
+			wsURL := strings.Replace(fmt.Sprint(*gatewayEndpoint, "/v2/websocket"), "http", "ws", 1)
+			gatewayWebsocket, _, err = websocket.DefaultDialer.Dial(wsURL, nil)
+			if err != nil {
+				return fmt.Errorf("WebSocket dial(%s): %v", wsURL, err)
+			}
+			endpoints["wss: gateway"] = gatewayWebsocket
+			endpoints["https: gateway"] = fmt.Sprint(*gatewayEndpoint, "/v2")
```

Comment:
Nit, maybe this would be more precise? We output these names, so might be worth adjusting.

```suggestion
			endpoints["ws(s): gateway"] = gatewayWebsocket
			endpoints["http(s): gateway"] = fmt.Sprint(*gatewayEndpoint, "/v2")
```
(same for sourcegraph endpoints below)